scripts/SREP/update-upgrade-policy-state/script.sh

```diff
@@ -26,15 +26,27 @@ get_cluster_uuid() {
 # Params:
 #   - Cluster UUID
 get_cluster_id() {
-    ocm list clusters --parameter search="external_id is '${1}'" --columns id --padding 20 --no-headers
+    oc get configmap cluster-config-v1 -n kube-system -o jsonpath='{.data.install-config}' | grep -oE 'api.openshift.com/id: [^ ]+' | awk '{print $2}'
 }
 
 # Get the UUID of the upgrade policy for the cluster.
 # Params:
 #   - Cluster ID
 get_policy_uuid() {
     local policy_uuid
-    policy_uuid=$(ocm get "/api/clusters_mgmt/v1/clusters/${1}/upgrade_policies/" | jq '.items[]? | .id' | tr -d '"')
+    local ocm_slug="api/clusters_mgmt/v1/clusters/${1}/upgrade_policies/"
+    if [[ -z "$env" ]]; then
```

Comment:
The script  (`if/else` clause) contains repeated code in both `get_policy_uuid()` and `update_upgrade_policy()`, better to define a reusable function to construct the API URL based on the `env` parameter.